src/main/java/seedu/address/model/entry/Date.java

```diff
@@ -3,17 +3,21 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 /**
  * Represents an Entry's date in the penny wise application.
  * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
  */
 public class Date {
     public static final String MESSAGE_CONSTRAINTS =
             "Dates should be of the format dd-mm-yyyy and it should only contain numbers";
+    public static final String VALIDATION_PATTERN = "dd-MM-yyyy";
     public static final String VALIDATION_REGEX =
             "^([0-2][0-9]||3[0-1])-(0[1-9]||[1-9]||1[0-2])-([0-9][0-9])?[0-9][0-9]$";
```

Comment:
Can the `VALIDATION_REGEX` can be removed already? Since now our dates are validated by the date pattern type.

Reply:
You're right, I will fix that, thanks!

Reply:
I think we can leave this first; I realized it's not straightforward to remove the `VALIDATION_REGEX` as there does not seem to be a direct way to validate using `VALIDATION_PATTERN` without using hacks like try-catch etc.